test/cql-pytest/test_secondary_index.py

```diff
@@ -64,3 +64,32 @@ def test_partition_order_with_si(cql, test_keyspace):
                 pass
             time.sleep(0.1)
         assert tokens2 == tokens
+
+# Test which ensures that indexes for a query are picked
+# by the order in which they appear in restrictions.
+# That way, users can deterministically pick which indexes
+# are used for which queries.
+# Ref: #7969
+@pytest.mark.xfail(reason="The order of picking indexes is currently arbitrary. Issue #7969")
+def test_order_of_indexes(scylla_only, cql, test_keyspace):
+    schema = 'p int primary key, v1 int, v2 int, v3 int'
+    with new_test_table(cql, test_keyspace, schema) as table:
+        cql.execute(f"CREATE INDEX my_v3_idx ON {table}(v3)")
+        cql.execute(f"CREATE INDEX my_v1_idx ON {table}(v1)")
+        cql.execute(f"CREATE INDEX my_v2_idx ON {table}((p),v2)")
+        # All queries below should use the first index they find
+        # in the list of restrictions. Tracing information will be
+        # consulted to ensure it's true
+        def index_used(query, index_name):
+            assert any([index_name in event.description for event in cql.execute(query, trace=True).get_query_trace().events])
```

Comment:
Very interesting use of tracing.
However, I'm (very) worried about one thing: How much time does it take for a traced query to complete? Could it wait for 2 seconds for each query like I saw in the past?
If this is the case, this test can take 20 seconds to complete :-( Could we run all 10 queries and only then go back to check their traces? 2 seconds is still a lot for a test, but it's better than 20.

Reply:
Theoretically, in order to wait for the tracing to be complete (which might take longer), one has to do the following:
```python
trace = .get_query_trace()
trace.populate(wait_for_complete=True)
```
I don't do that here, piggy-backing on the fact that this query requested tracing with the protocol flag, which in Scylla means that the tracing tables are populated right away. Ref:
https://github.com/scylladb/scylla/blob/5109bf8b996ccd6046590e51ca451027c642e26b/transport/server.cc#L706-L708

So, the tracing info is actually instantly available, contrary to, for instance, probabilistic tracing, which might have a delay due to buffering.

Reply:
Excellent!